### DISCUSSION TITLE
MQTT und Forecast Fehler nach DNS change in Fritzbox

### DISCUSSION THREAD

**Discussion by jojunker** [solectrus/solectrus#2182] [Q&A] [answered]
**Was ist passiert (oder was glaube ich ist passiert)**
Ich habe die DNS Adresse in meiner Fritzbox anpassen müssen um meinen defekten pihole zu umgehen.
Das hat soweit für alle Geräte im Netzwerk funktioniert.

Mein Raspberry mit Solectrus im Docker will nach einem Reboot aber nicht mehr.
Fehler:

**docker ps**

```
CONTAINER ID   IMAGE                                         COMMAND                  CREATED       STATUS                         PORTS      NAMES
11faacae6657   ghcr.io/solectrus/solectrus:latest            "docker/startup.sh"      8 hours ago   Restarting (1) 4 seconds ago              solectrus-app-1

baba0c821a76   ghcr.io/solectrus/mqtt-collector:latest       "/bin/sh -c 'bundle …"   8 hours ago   Restarting (1) 4 seconds ago              solectrus-mqtt-collector-1

e129849e6625   ghcr.io/solectrus/forecast-collector:latest   "/bin/sh -c 'bundle …"   8 hours ago   Restarting (1) 4 seconds ago              solectrus-forecast-collector-1

1b6b4164f959   postgres:15-alpine                            "docker-entrypoint.s…"   8 hours ago   Up 8 minutes (healthy)         5432/tcp   solectrus-db-1

abfdc8926d92   influxdb:2.7-alpine                           "/entrypoint.sh infl…"   8 hours ago   Up 8 minutes (healthy)         8086/tcp   solectrus-influxdb-1

95eb2b491c5b   redis:7-alpine                                "docker-entrypoint.s…"   8 hours ago   Up 8 minutes (healthy)         6379/tcp   solectrus-redis-1
```

**docker logs solectrus-app-1**

Starting SOLECTRUS...
Version: v0.12.0 - 2023-07-08T11:05:59+02:00
----------------
```
influxdb (172.19.0.5:8086) open
InfluxDB is up and running!
db (172.19.0.7:5432) open
PostgreSQL is up and running!
Preparing database...
/usr/local/lib/ruby/3.2.0/error_highlight.rb:1:in `require_relative': /usr/local/lib/ruby/3.2.0/error_highlight/base.rb:305: syntax error, unexpected end-of-input, expecting `end' or dummy end (SyntaxError)
    deP9
        ^

        from /usr/local/lib/ruby/3.2.0/error_highlight.rb:1:in `<top (required)>'
        from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:gem_prelude>:10:in `<internal:gem_prelude>'
Database is ready!
/usr/local/lib/ruby/3.2.0/error_highlight.rb:1:in `require_relative': /usr/local/lib/ruby/3.2.0/error_highlight/base.rb:305: syntax error, unexpected end-of-input, expecting `end' or dummy end (SyntaxError)
    deP9
        ^

        from /usr/local/lib/ruby/3.2.0/error_highlight.rb:1:in `<top (required)>'
        from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from <internal:gem_prelude>:10:in `<internal:gem_prelude>'


```

MQTT und Forecast scheinen betroffen zu sein und damit stoppt bzw. restartet solectrus-app-1 immer wieder...

Ich hoffe ich muss nur einen Cache löschen oder aktualisieren.... hab aber leider keine Idee wo ich anfangen soll.
Danke

**Comment by ledermann** [comment, marked as answer]
Seltsames Problem, das ist mir noch nie begegnet. Betroffen sind die drei Docker-Container, die auf Ruby basieren. Bemängelt wird in ein Syntax-Error in einer Ruby-Standard-Bibliothek. Sind die Images möglicherweise beschädigt? Ist aber unwahrscheinlich, dass es gleich drei Container betrifft.

Vielleicht mal die Images löschen und neu holen lassen? Das geht wie folgt:

```bash
docker compose down
docker rmi ghcr.io/solectrus/solectrus:latest 
docker rmi ghcr.io/solectrus/senec-collector:latest 
docker rmi ghcr.io/solectrus/forecast-collector:latest
docker system prune
docker compose up 
```

An einer DNS-Anpassung kann das eher nicht liegen, zumindest sehe ich da keinen Zusammenhang. Vielleicht wurde durch den Reboot etwas ganz anderes verändert - ich habe allerdings keine Vermutung, was den oben angegeben Fehler auslösen könnte.

**Reply by jojunker** [reply]
@ledermann : Vielen Dank für Deine schnelle Reaktion! Docker ist für mich noch neu - die Anleitung hat perfekt geholfen.
Nach docker compose up läuft alles wieder wie vorher - alle Daten da :-)
Frage - gibt es evtl. ein Logfile, dass Aufschluss über den Fehler bzw. die Ursache geben könnte?

**Reply by ledermann** [reply]
Freut mich, dass es geklappt hat, auch wenn die Ursache für mich unklar bleibt. Logfiles sind mit dem `docker system prune` leider alle gelöscht. Ok, hätte man natürlich vorher mal vollständig untersuchen können ...

Noch zu Docker: Da Images grundsätzlich unveränderlich sind (im Gegensatz zu Containern), wäre ein erneutes Herunterladen unter normalen Umständen nicht erforderlich gewesen. Wenn aber genau die drei Container kaputtgehen, die auf dem gleichen Ruby-Image basieren, dann sieht mir das noch am ehesten nach einem Storage-Problem aus.

Vielleicht behältst Du den Raspberry in der nächsten Zeit mal genauer im Auge, möglicherweise passieren weitere komische Dinge. Insbesondere die SD-Karte hält ja nicht ewig.
